pkg/multiclustermanager/routes.go

```diff
@@ -93,11 +93,20 @@ func router(ctx context.Context, localClusterEnabled bool, tunnelAuthorizer *mcm
 	unauthed.PathPrefix("/v3-public").Handler(publicAPI)
 
 	// Authenticated routes
-	authed := mux.NewRouter()
-	authed.UseEncodedPath()
 	impersonatingAuth := auth.ToMiddleware(requests.NewImpersonatingAuth(sar.NewSubjectAccessReview(clusterManager)))
+	saAuth := auth.ToMiddleware(requests.NewServiceAccountAuth(scaledContext, clustermanager.ToRESTConfig))
 	accessControlHandler := rbac.NewAccessControlHandler()
 
+	saauthed := mux.NewRouter()
+	saauthed.UseEncodedPath()
+	saauthed.PathPrefix("/k8s/clusters/{clusterID}/apis/authentication.k8s.io/v1/tokenreviews").Handler(k8sProxy)
```

Comment:
This is deliberately limited to tokenreviews route to minimize the blast radius while working on this. We idea is to extend this to the whole `/k8s/clusters` once we come up with a solid implementation.